src/analysis/fourier.lean

```diff
@@ -53,29 +58,155 @@ this Hilbert basis.
+open real complex topological_space continuous_map measure_theory measure_theory.measure
+  algebra submodule set interval_integral
 
 /-! ### Choice of measure on the circle -/
 
-section haar_circle
-/-! We make the circle into a measure space, using the Haar measure normalized to have total
-measure 1. -/
+section circle_measure
+/-! We make the circle into a measure space, using the measure transported from `(0, 2 * π]`,
+ normalized to have total measure 1. -/
 
 instance : measurable_space circle := borel circle
 instance : borel_space circle := ⟨rfl⟩
 
-/-- Haar measure on the circle, normalized to have total measure 1. -/
-@[derive is_haar_measure]
-def haar_circle : measure circle := haar_measure ⊤
+lemma measurable_arg' : measurable arg' :=
+begin
+  let t1 := { z : ℂ | 0 < arg z}, let t2 := t1ᶜ,
+  apply measurable_of_measurable_union_cover t1 t2 _ _ (by simp),
+  { have : (λ (a : t1), arg' a) = (λ (a : t1), arg a),
+    { ext, rw arg', split_ifs, refl, exfalso, exact h x.property }, rw this,
+    exact measurable_arg.comp measurable_subtype_coe },
+  { have : (λ (a : t2), arg' a) = (λ (a : t2), arg a + (2 * π)),
+    { ext, rw arg', split_ifs, exfalso, exact x.property h, refl, }, rw this,
+    exact (measurable_arg.add_const _).comp measurable_subtype_coe },
+  { exact measurable_set_lt measurable_const measurable_arg },
+  { exact (measurable_set_lt measurable_const measurable_arg).compl },
+end
+
+/-- Measurable equivalence from `Ioc 0 (2 * π)` to `circle`, whose underlying set bijection is
+`circle.circle_equiv`. -/
+def circle_m_equiv : measurable_equiv (Ioc 0 (2 * π)) circle :=
+{ measurable_inv_fun := by
+  { rw circle.circle_equiv, rw circle.arg'_equiv,
+    simp only [equiv.inv_fun_as_coe, equiv.symm_symm, equiv.coe_fn_mk, equiv.coe_fn_symm_mk],
+    rw ←(measurable_embedding.subtype_coe
+          (@measurable_set_Ioc ℝ _ _ _ _ _ _ _)).measurable_comp_iff,
+    exact measurable_arg'.comp continuous_subtype_coe.measurable },
+  measurable_to_fun := (exp_map_circle.continuous.borel_measurable).comp measurable_subtype_coe,
+  .. circle.circle_equiv }
+
+lemma arg'_m_emb : measurable_embedding (arg' ∘ coe : circle → ℝ) :=
+begin
+  convert (measurable_embedding.subtype_coe measurable_set_Ioc).comp
+    (circle_m_equiv.symm).measurable_embedding using 1,
+end
+
+/-- Measure on the circle, normalized to have total measure 1. -/
+def circle_measure : measure circle :=
+  (ennreal.of_real (1 / (2 * π)) • volume).map circle_m_equiv
+
+lemma circle_measure_univ : circle_measure univ = 1 :=
+begin
+  dsimp only [circle_measure],
+  rw [circle_m_equiv.map_apply, preimage_univ, measure.smul_apply, id.smul_eq_mul,
+    ←volume_image_subtype_coe (@measurable_set_Ioc ℝ _ _ _ _ _ _ _), image_univ,
+    subtype.range_coe, real.volume_Ioc, ←ennreal.of_real_mul (one_div_nonneg.mpr two_pi_pos.le)],
+  ring_nf, field_simp [real.pi_ne_zero],
+end
+
+instance : is_probability_measure circle_measure := ⟨circle_measure_univ⟩
```

Comment:
I think it would be more elegant simply to make this
```suggestion
instance : is_probability_measure circle_measure :=
⟨begin
  dsimp only [circle_measure],
  rw [circle_m_equiv.map_apply, preimage_univ, measure.smul_apply, id.smul_eq_mul,
    ←volume_image_subtype_coe (@measurable_set_Ioc ℝ _ _ _ _ _ _ _), image_univ,
    subtype.range_coe, real.volume_Ioc, ←ennreal.of_real_mul (one_div_nonneg.mpr two_pi_pos.le)],
  ring_nf, field_simp [real.pi_ne_zero],
end⟩
```